drivers/dma/dma_mcux_lpc.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2020 NXP Semiconductor INC.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common part of DMA drivers for some NXP SoC.
+ */
+#include <device.h>
+#include <soc.h>
+#include <drivers/dma.h>
+#include <fsl_dma.h>
+#include <fsl_inputmux.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT nxp_lpc_dma
+#define LPC_MCUX_NUM_DMA_CHANNELS 32
```

Comment:
This should be imported from **[device]_features.h** because other devices may have a different number of channels (e.g. LPC55S69 has 23).
`FSL_FEATURE_DMA_NUMBER_OF_CHANNELS` is the macro; **[device]_features.h** is imported when `#include <fsl_device_registers.h>`

Reply:
This is a good point. I have updated the implementation to pick the number of channels from the dts file. Also the DMA data structures are allocated dynamically via k_malloc to reduce memory waste.